tests/errors/terror_in_if.test

```diff
@@ -0,0 +1,7 @@
+discard """
+  description: "Either the if or else branch has an error"
+  reject: true
+"""
+(If (Ident "true")
+  (Ident "missing")
+  (IntVal 10))
```

Comment:
Given the recent issues that were surface, it's probably a good idea to make the testing exhausive by putting an error in the else branch as a separate test.

The other test can be a simple reversal:

```nim
iscard """
  description: "Either the if or else branch has an error"
  reject: true
"""
(If (Ident "true")
  (IntVal 10))
  (Ident "missing")
```